### ISSUE TITLE
How to create the private key?

### ISSUE THREAD

**Issue by shinhwasun** [blue-prism/SSH.NET#1]
Hi team

the VBO wrapper of the SSH.NET is very fit for my business. I would like to connect with a remote linux server via SSH and just run a command "df" to get the disk space of the remote server.

But I have difficulty to create the private key. I tried to create private key by the puttygen and put it to the folder "C:\Program Files\Blue Prism Limited\Blue Prism Automate". But while executing the process. it told me the private key file was invalid.

So here I raise this ticket to ask:
1. Why asking for private as I have provided the user name and password for the remote server.
2. What is the valid private key file like? anyway to create it?
3. Do we need to append the public key to the file "authorized_keys" of the remote server?

Thanks and best regard

And The key file I generated was a .ppk file
PuTTY-User-Key-File-3: ssh-rsa
Encryption: aes256-cbc
Comment: rsa-key-20210615
Public-Lines: 6
AAAAB3NzaC1yc2EAAAADAQABAAABAQDTNIeTkFG5Po03Zh0x15sqZtUzmAedGoNp
c5KdTHmA/a5C8pjZAvDgkA1niyWc90K+1T8ymrgBEdH3LY64vFWh0dmgQsgI9ZT1
NF2HsuKVuppwffbT7SWEDUKpd7DstNdRAnFp38ZxJTjls12EIbqt7pSbgE6McuC0
LV1RQ6ypTQrXH943HgtT5q5aosSvZaSFGgEkYjsTQk5mZkxsvXlvZqy/ZPH5Y0Ys
KGVEXaH6KkS2B4Yfa5f4ZYZi8fs9klQv2FAhmMCFkqeChGDjLsKJzaKnLfTHxvxZ
bgBinXt4y9YsPSVG3bCDmy7MbLjRJ2cCDq0RHko0RO20I6X1wWhP
Key-Derivation: Argon2id
Argon2-Memory: 8192
Argon2-Passes: 5
Argon2-Parallelism: 1
Argon2-Salt: 12ef564ae2209ee4a50874b45e23d3ce
Private-Lines: 14
aObo2pwinISukVtVHufxvFcxRRFNiWf8X6+6gCMcHsnt2atFv9zw1Y1iD0MNAfTY
9lmE1esV/pJZ2q8PTq2sifQ1R3cHyjvwPL/f+5il59ej1Bqf2z+0yhGWTEJhu6kO
7TwiDdCaLQk1QU3QLHN2/CMoEpUigoaRgFmLULvTYSMBxZDNbcpNwGU4tR+Kbp7d
3e0MzsFlE2aJDlrmsjSeCVV7zdb4nATMZ7DUS82KD2Yb+Z2h+h2DKHbyOK/86714
VBLOKHOjMvsLiNxqkzq1Cl7rxRYkM+qIhG6RnxvoQz0XRWyZitw+MiFYqtqtzrKK
tsmLLtywrJNb9nI/GabnVhehkjDFl4kggv+2Zo8FGuhIqN/RdOx9snbX1+c3AS6z
xQuj8yLmcKwwU1l0hV87z0BUQ15Ooi/jByJulZn3dEcl0iwbruCTb2nZMPXKH1Se
Jgd4xeow/9fbe7Y+01ny6OGiLhjFPibGAmPegddsd+ywUigFi/GvAdH6g44BzOiy
NYKT+oGf3EJ/xxdb2K8kEJZU/+kCn7q+XqLLg5S5gh+e3F4bG/r/XY3E+kbbt6CI
Y0PWrfV4XQKyEvocmLLnBAyrrMGkylfF+leUxxOnjddhHjHj+D14HM9Lxz7sx7JC
3vdfTWUfVzkpCUX432OYduvHP/YGMnDjH8pNnbJ2rV5Upiaw+ny2x+Yo/TGnYVDw
vdoz167o53S4s6WtGLxXbrTYZLF5fbnXG4gnylr2uwoG34h/wJbSwnShXlljHolA
iF7I6CrEH4SYyRFIwdOlwVEz1VbYfIxeWTGc9ipYCKwrBkAlbFT1XGA/nUwwo/Ac
f1xBw1BLzT25vxFh+TisvTXfjrOrzTSFIPRvo5ogngxeIbgJXi4cdNheQfFb5dkD
Private-MAC: 20ca4bd607dfee4185d045e67fc408f667cdd1db411e01eb88ccc348b62f7b21

**Comment by shashankkumar88** [comment]
@shinhwasun PrivateKey is optional. If you have already provided username and password you need not provide the PrivateKey.

**Comment by shinhwasun** [comment]
Many thanks Shashank for your fast responding. And yes, I tried leaving the input field "Private Key FilePath" empty. 
However exception happens when executing: "No suitable authentication method found to complete authentication (publickey,keyboard-interactive)." Anything I need to do at the remote server side?
![image](https://user-images.githubusercontent.com/85915235/121989560-9f87c700-cd9c-11eb-9b9a-5450b2a2fc2e.png)

**Comment by shashankkumar88** [comment]
It seems the server is expecting KeyboardInteractiveAuthenticationMethod rather than PasswordAuthenticationMethod. You might need to add some custom code as mentioned here https://github.com/sshnet/SSH.NET/issues/541#issuecomment-507103900 or https://stackoverflow.com/questions/34875331/ssh-net-no-suitable-authentication-method-found

**Comment by shinhwasun** [comment]
Hi Shashank
I finally generated the correct private key file and resolved my problem.
I followed this guide to generate the private key and public key: https://www.ssh.com/academy/ssh/putty/windows/puttygen
The public key should be append to the file /.ssh/authorized_keys of the remote server and the private should be put in the folder C:\Program Files\Blue Prism Limited\Blue Prism Automate on the user machine.
Notice: while saving the private, DON'T directly click the button "Save private key" but do Conversations -> export OpenSSH Key.
